src/expr.cpp

```diff
@@ -400,16 +400,11 @@ Expr *Expr::PartialWrt(hParam p) const {
     ssassert(false, "Unexpected operation");
 }
 
-void Expr::ParamsUsedList(std::vector<hParam> *list) const {
+void Expr::ParamsUsedList(ParamSet *list) const {
```

Comment:
Have you really tested it? There is no "large" param sets actually exist, linear search is the best suitable for this kind of data.

Reply:
Admittedly no expression currently contains more than a handful of parameters, and even less duplicated ones in the largest of expressions, but in my testing it made no difference in performance one way or the other even with models that generate a huge amount of equations, and the reason I went with an unordered set was simply because it made the code simpler since it doesn't require an explicit check for duplicates. I'm fine with dropping this change if the maintainers feel uncomfortable with it, but as noted above this was done as a cleanup rather than as an optimisation.